src/plugins/memdump/memdump.cpp

```diff
@@ -873,6 +874,173 @@ static event_response_t set_information_thread_hook_cb(drakvuf_t drakvuf, drakvu
     return VMI_EVENT_RESPONSE_NONE;
 }
 
+struct exec_fault_data
+{
+    memdump* plugin;
+    addr_t rip;
+};
+
+struct access_fault_result_t: public call_result_t
+{
+    access_fault_result_t() : call_result_t(), fault_va() {}
+
+    addr_t fault_va;
+};
+
+std::set<std::pair<addr_t, addr_t>> known_vads;
```

Comment:
Does this has to be sorted? If no, then how about using std::unordered_set?

Reply:
hint `std::set<std::pair<>>` is just `std::map` :)
(even better `std::unordered_map).

Reply:
I will include this

Reply:
After internal discussion we came to conclusion that `std::unordered_set<std::pair>` is correct here